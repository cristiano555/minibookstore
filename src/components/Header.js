import React from 'react';
import '../index.css';

class Header extends React.Component {

constructor(){

    super();
    this.state = {
        bookstoreName: "Black Books",
       clicked: true

    }
}
/*
f = () => {
    this.setState({
        clicked: false
    })
}
    */

changeColor = () => {

if(this.state.clicked) {

    this.setState({
        bookstoreName: "Red Books"
    })
} else {
    this.setState({
        bookstoreName: "Black Books"
    })
}

    this.setState({
        clicked: !this.state.clicked
    })

}




    render() {
        return (
            <div class="row header col-xs-12">
             <button onClick={this.changeColor}>Click</button>   <h1>{this.state.bookstoreName}</h1>
            </div>

        ) 
    }
}


export default Header;