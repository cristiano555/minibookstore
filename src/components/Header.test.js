import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe('Header tests', () => {
  

  it('Header renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Hello world renders', () => {
  const wrapper = shallow(<Header/>);
  console.log(wrapper.debug());
 // expect(wrapper.find('div').text()).toBe('Header div');
  })

  it('Snapshot matches renders', () => {
    const wrapper = shallow(<Header/>);
    expect(wrapper).toMatchSnapshot();
    })

})