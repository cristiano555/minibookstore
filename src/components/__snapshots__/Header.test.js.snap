// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests Snapshot matches renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          onClick={[Function]}
        >
          Click
        </button>,
        "   ",
        <h1>
          Black Books
        </h1>,
      ],
      "class": "row header col-xs-12",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Click",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Click",
        "type": "button",
      },
      "   ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Black Books",
        },
        "ref": null,
        "rendered": "Black Books",
        "type": "h1",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            onClick={[Function]}
          >
            Click
          </button>,
          "   ",
          <h1>
            Black Books
          </h1>,
        ],
        "class": "row header col-xs-12",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Click",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Click",
          "type": "button",
        },
        "   ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Black Books",
          },
          "ref": null,
          "rendered": "Black Books",
          "type": "h1",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
